connector/src/main/java/org/geysermc/connector/utils/InventoryUtils.java

```diff
@@ -68,19 +69,34 @@ public static void openInventory(GeyserSession session, Inventory inventory) {
     public static void closeInventory(GeyserSession session, int windowId) {
         if (windowId != 0) {
             Inventory inventory = session.getInventoryCache().getInventories().get(windowId);
-            if (inventory != null) {
+            Inventory openInventory = session.getInventoryCache().getOpenInventory();
+            session.getInventoryCache().uncacheInventory(windowId);
+            if (inventory != null && openInventory != null && inventory.getId() == openInventory.getId()) {
                 InventoryTranslator translator = Translators.getInventoryTranslators().get(inventory.getWindowType());
                 translator.closeInventory(session, inventory);
-                session.getInventoryCache().uncacheInventory(windowId);
                 session.getInventoryCache().setOpenInventory(null);
+            } else {
+                return;
             }
         } else {
             Inventory inventory = session.getInventory();
             InventoryTranslator translator = Translators.getInventoryTranslators().get(inventory.getWindowType());
             translator.updateInventory(session, inventory);
         }
+
         session.setCraftSlot(0);
         session.getInventory().setCursor(null);
+        updateCursor(session);
+    }
+
+    public static void closeWindow(GeyserSession session, int windowId) {
+        //Spamming close window packets can bug the client
+        if (System.currentTimeMillis() - session.getLastWindowCloseTime() > 500) {
+            ContainerClosePacket closePacket = new ContainerClosePacket();
+            closePacket.setWindowId((byte) windowId);
+            session.sendUpstreamPacket(closePacket);
+            session.setLastWindowCloseTime(System.currentTimeMillis());
+        }
     }
```

Comment:
Instead of discarding them, would it not be better just delaying them? (assuming there a different window id) Something like this is how other parts of Geyser do it.
```
session.getConnector().getGeneralThreadPool().schedule(() -> session.getEntityCache().spawnEntity(entity), 100, TimeUnit.MILLISECONDS)
```